### PR TITLE
Adjust kvbc command handler to pre-execution

### PR DESCRIPTION
**Pull request by yontyon**
In order to perform tests, we need to adjust the KVBCReplica command handler to pre-execution.
This PR handles it as well as hardening correlation id in pre-execution.

### CODE REVIEW

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 1]
```diff
@@ -234,12 +234,11 @@ void PreProcessor::onMessage<PreProcessReplyMsg>(PreProcessReplyMsg *msg) {
     clientEntry->clientReqInfoPtr->handlePreProcessReplyMsg(preProcessReplyMsg);
     result = clientEntry->clientReqInfoPtr->getPreProcessingConsensusResult();
   }
-  MDC_CID_PUT(GL, clientEntry->clientReqInfoPtr->getPreProcessRequest()->getCid());
   switch (result) {
     case CONTINUE:  // Not enough equal hashes collected
       return;
     case COMPLETE:  // Pre-processing consensus reached
-      finalizePreProcessing(clientId, reqSeqNum);
+      finalizePreProcessing(clientId, reqSeqNum, "" /* cid will be added later */);
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp]: Please move the comment out from the parentheses.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 2]
```diff
@@ -281,12 +280,13 @@ void PreProcessor::finalizePreProcessing(NodeIdType clientId, SeqNum reqSeqNum)
...
                                                      clientEntry->clientReqInfoPtr->getMyPreProcessedResultLen(),
-                                                     clientEntry->clientReqInfoPtr->getMyPreProcessedResult());
+                                                     clientEntry->clientReqInfoPtr->getMyPreProcessedResult(),
+                                                     cid);
   }
   incomingMsgsStorage_->pushExternalMsg(move(clientRequestMsg));
   preProcessorMetrics_.requestSentForFurtherProcessing.Get().Inc();
   releaseClientPreProcessRequest(clientId, reqSeqNum);
-  LOG_DEBUG(GL, "Pre-processing completed for clientId=" << clientId << " reqSeqNum=" << reqSeqNum);
+  LOG_INFO(GL, "Pre-processing completed for clientId=" << clientId << " reqSeqNum=" << reqSeqNum);
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp]: Please return it back to the debug level.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 3]
```diff
@@ -297,6 +297,7 @@ void PreProcessor::registerClientPreProcessRequest(ClientPreProcessReqMsgUniqueP
   {
     // Only one request is supported per client for now
     auto &clientEntry = ongoingRequests_[clientId];
+    LOG_INFO(GL, "bug(1)");
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp]: Please remove all the logs you added for debugging.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 4]
```diff
@@ -330,7 +331,7 @@ void PreProcessor::handleClientPreProcessRequest(ClientPreProcessReqMsgUniquePtr
                                                                                          clientReqMsg->getCid());
   registerClientPreProcessRequest(move(clientReqMsg));
   sendPreProcessRequestToAllReplicas(preProcessRequestMsg);
-
+  LOG_INFO(GL, "bug(2)");
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp]: As before.

File: bftengine/src/preprocessor/PreProcessor.cpp [hunk 5]
```diff
@@ -405,6 +410,8 @@ void PreProcessor::handlePreProcessedReqByNonPrimary(uint16_t clientId, ReqId re
 void PreProcessor::handleReqPreProcessingJob(PreProcessRequestMsgSharedPtr preProcessReqMsg,
                                              bool isPrimary,
                                              bool isRetry) {
+  MDC_CID_PUT(GL, preProcessReqMsg->getCid());
+  LOG_INFO(GL, "bug(3)");
```
Comment (yuliasherman) [inline bftengine/src/preprocessor/PreProcessor.cpp]: Remove

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 1 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 2 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 3 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 4 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

File: bftengine/src/preprocessor/PreProcessor.cpp (hunk 5 above)
Comment (yontyon) [reply bftengine/src/preprocessor/PreProcessor.cpp]: done

**Pull request merged**